source/extensions/quic_listeners/quiche/BUILD

```diff
@@ -404,11 +405,12 @@ envoy_cc_extension(
 envoy_cc_extension(
     name = "quic_factory_lib",
     category = (
-        "envoy.transport_sockets.downstream",
-        "envoy.transport_sockets.upstream",
+        # Remove QUIC as an extension
```

Comment:
Is there value in doing this before we have the build option?

Reply:
You mean doing what I'm doing in this PR? Mainly I just wanted to get all the config hidden to make it clear it's alpha / going to be changed / probably going away, and I had to deal with a lot of it for the docs in this PR. I can split it out for sure but would rather just land it if you are OK with it in general.